drivers/video/gc2145.c

```diff
@@ -0,0 +1,1150 @@
+	{0xf2, 0x0f},
+	{0xfe, 0x02},
+	{0x40, 0xbf},
+	{0x46, 0xcf},
+	{0xfe, 0x00},
+
+	{0xfe, 0x00},
+	{0x05, 0x01},
+	{0x06, 0x1C},
+	{0x07, 0x00},
+	{0x08, 0x32},
+	{0x11, 0x00},
+	{0x12, 0x1D},
+	{0x13, 0x00},
+	{0x14, 0x00},
+
+	{0xfe, 0x01},
+	{0x3c, 0x00},
+	{0x3d, 0x04},
+	{0xfe, 0x00},
+	{0x00, 0x00},
+};
+
+struct gc2145_config {
+	struct i2c_dt_spec i2c;
+#if DT_INST_NODE_HAS_PROP(0, reset_gpios)
+	struct gpio_dt_spec reset_gpio;
+#endif
+};
+
+struct gc2145_data {
+	struct video_format fmt;
+};
+
+#define GC2145_VIDEO_FORMAT_CAP(width, height, format) \
+	{                                              \
+	.pixelformat = format,                         \
+	.width_min = width,                            \
+	.width_max = width,                            \
+	.height_min = height,                          \
+	.height_max = height,                          \
+	.width_step = 0,                               \
+	.height_step = 0,                              \
+	}
+
+static const struct video_format_cap fmts[] = {
+	GC2145_VIDEO_FORMAT_CAP(1600, 1200, VIDEO_PIX_FMT_RGB565), /* UXGA  */
+	{0},
+};
+
+static int gc2145_write_reg(const struct i2c_dt_spec *spec, uint8_t reg_addr, uint8_t value)
+{
+	int ret;
+	uint8_t tries = 3;
+
+	/*
+	 * It rarely happens that the camera does not respond with ACK signal.
+	 * In that case it usually responds on 2nd try but there is a 3rd one
+	 * just to be sure that the connection error is not caused by driver
+	 * itself.
+	 */
+	do {
+		ret = i2c_reg_write_byte_dt(spec, reg_addr, value);
+		if (!ret) {
+			return 0;
+		}
+		/* If writing failed wait 5ms before next attempt */
+		k_msleep(5);
+
+	} while (tries--);
+
+	LOG_ERR("failed to write 0x%x to 0x%x,", value, reg_addr);
+	return ret;
+}
+
+static int gc2145_read_reg(const struct i2c_dt_spec *spec, uint8_t reg_addr, uint8_t *value)
+{
+	int ret;
+	uint8_t tries = 3;
+
+	/*
+	 * It rarely happens that the camera does not respond with ACK signal.
+	 * In that case it usually responds on 2nd try but there is a 3rd one
+	 * just to be sure that the connection error is not caused by driver
+	 * itself.
+	 */
+	do {
+		ret = i2c_reg_read_byte_dt(spec, reg_addr, value);
+		if (!ret) {
+			return 0;
+		}
+		/* If writing failed wait 5ms before next attempt */
+		k_msleep(5);
+
+	} while (tries--);
+
+	LOG_ERR("failed to read 0x%x register", reg_addr);
+	return ret;
+}
+
+static int gc2145_write_all(const struct device *dev, const struct gc2145_reg *regs,
+			    uint16_t reg_num)
+{
+	const struct gc2145_config *cfg = dev->config;
+
+	for (uint16_t i = 0; i < reg_num; i++) {
+		int ret;
+
+		ret = gc2145_write_reg(&cfg->i2c, regs[i].addr, regs[i].value);
+		if (ret < 0) {
+			return ret;
+		}
+	}
+
+	return 0;
+}
+
+static int gc2145_soft_reset(const struct device *dev)
+{
+	int ret;
+	const struct gc2145_config *cfg = dev->config;
+
+	/* Initiate system reset */
+	ret = gc2145_write_reg(&cfg->i2c, GC2145_REG_RESET, GC2145_REG_SW_RESET);
+
+	k_msleep(300);
```

Comment:
Pointing at it in case you wanted to convert all occurrences of `k_msleep()` to `k_sleep(K_MSEC())`...

Maybe this is where the comment block belongs the most, as it is the longest delay?